sphinx/domains/__init__.py

```diff
@@ -201,6 +201,11 @@ class Domain:
     data: dict
     #: data version, bump this when the format of `self.data` changes
     data_version = 0
+    #: Value for an empty inventory of objects for this domain.
+    #: It must be copy.deepcopy-able.
+    #: If this value is overridden, then the various intersphinx methods in the domain should
+    #: probably also be overridden.
+    initial_intersphinx_inventory: dict = {}
```

Comment:
Could we get a better type annotation?

Reply:
I have added another sentence to the description to clarify that it is up to the concrete domain to design the dictionary, and that intersphinx won't look into it.

Reply:
Ah, actually I wasn't sure if mypy wouldn't complain if you don't use something like `dict[Any, Any]`. However. you could perhaps make it a ClassVar?

Reply:
You mean as in declaring the type to be ``ClassVar[dict]``? Non of the other attributes have this. Perhaps they should?

Reply:
Err... probably, unless they could be instance-specific. But I think domains should be considered as singletons (even if multiple instances could exist, there should only be one instance of a specific domain at any time (assuming no parallelism)) so it would capture the semantic a bit better IMO.

I think we didn't type them because this syntax did not exist yet / was not well-supported but if we add new things, I'd suggest that those new things are "clean". And in a separate PR, we cleanup the old things (rather than having a new thing in an old-style).